The job **`mingw-check`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/10530912028/job/29181863993) [(plain)](https://github.com/rust-lang/rust/commit/d9c724bae1e0efe58ab3bb9f466e67c348dc849e/checks/29181863993/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
#16 2.723 Building wheels for collected packages: reuse
#16 2.724   Building wheel for reuse (pyproject.toml): started
#16 2.973   Building wheel for reuse (pyproject.toml): finished with status 'done'
#16 2.974   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132715 sha256=dfa09868353292d98f811d3efdb0d54d07389e808efc71d68e3b93c514bf8bec
#16 2.975   Stored in directory: /tmp/pip-ephem-wheel-cache-jf5ax76h/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#16 2.977 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#16 3.376 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#16 3.376 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#16 DONE 3.5s
---
---- core/src/num/f16.rs - f16::f16::clamp (line 1258) stdout ----
error: multiple `cfg` predicates are specified
##[error] --> core/src/num/f16.rs:1260:31
  |
5 | #[cfg(target_arch = "x86_64", target_os = "linux")] {

error: aborting due to 1 previous error

Couldn't compile the test.
Couldn't compile the test.
---- core/src/num/f16.rs - f16::f16::classify (line 410) stdout ----
error: multiple `cfg` predicates are specified
##[error] --> core/src/num/f16.rs:412:31
  |
5 | #[cfg(target_arch = "x86_64", target_os = "linux")] {

error: aborting due to 1 previous error

Couldn't compile the test.
Couldn't compile the test.
---- core/src/num/f16.rs - f16::f16::from_be_bytes (line 1072) stdout ----
error: multiple `cfg` predicates are specified
##[error] --> core/src/num/f16.rs:1074:31
  |
5 | #[cfg(target_arch = "x86_64", target_os = "linux")] {

error: aborting due to 1 previous error

Couldn't compile the test.
Couldn't compile the test.
---- core/src/num/f16.rs - f16::f16::from_bits (line 959) stdout ----
error: multiple `cfg` predicates are specified
##[error] --> core/src/num/f16.rs:961:31
  |
5 | #[cfg(target_arch = "x86_64", target_os = "linux")] {

error: aborting due to 1 previous error

Couldn't compile the test.
Couldn't compile the test.
---- core/src/num/f16.rs - f16::f16::from_le_bytes (line 1095) stdout ----
error: multiple `cfg` predicates are specified
##[error] --> core/src/num/f16.rs:1097:31
  |
5 | #[cfg(target_arch = "x86_64", target_os = "linux")] {

error: aborting due to 1 previous error

Couldn't compile the test.
Couldn't compile the test.
---- core/src/num/f16.rs - f16::f16::from_ne_bytes (line 1125) stdout ----
error: multiple `cfg` predicates are specified
##[error] --> core/src/num/f16.rs:1127:31
  |
5 | #[cfg(target_arch = "x86_64", target_os = "linux")] {

error: aborting due to 1 previous error

Couldn't compile the test.
Couldn't compile the test.
---- core/src/num/f16.rs - f16::f16::is_finite (line 320) stdout ----
error: multiple `cfg` predicates are specified
##[error] --> core/src/num/f16.rs:322:31
  |
5 | #[cfg(target_arch = "x86_64", target_os = "linux")] {

error: aborting due to 1 previous error

Couldn't compile the test.
Couldn't compile the test.
---- core/src/num/f16.rs - f16::f16::is_infinite (line 294) stdout ----
error: multiple `cfg` predicates are specified
##[error] --> core/src/num/f16.rs:296:31
  |
5 | #[cfg(target_arch = "x86_64", target_os = "linux")] {

error: aborting due to 1 previous error

Couldn't compile the test.
Couldn't compile the test.
---- core/src/num/f16.rs - f16::f16::is_nan (line 262) stdout ----
error: multiple `cfg` predicates are specified
##[error] --> core/src/num/f16.rs:264:31
  |
5 | #[cfg(target_arch = "x86_64", target_os = "linux")] {

error: aborting due to 1 previous error

Couldn't compile the test.
Couldn't compile the test.
---- core/src/num/f16.rs - f16::f16::is_normal (line 378) stdout ----
error: multiple `cfg` predicates are specified
##[error] --> core/src/num/f16.rs:380:31
  |
5 | #[cfg(target_arch = "x86_64", target_os = "linux")] {

error: aborting due to 1 previous error

Couldn't compile the test.
Couldn't compile the test.
---- core/src/num/f16.rs - f16::f16::is_subnormal (line 348) stdout ----
error: multiple `cfg` predicates are specified
##[error] --> core/src/num/f16.rs:350:31
  |
5 | #[cfg(target_arch = "x86_64", target_os = "linux")] {

error: aborting due to 1 previous error

Couldn't compile the test.
Couldn't compile the test.
---- core/src/num/f16.rs - f16::f16::to_bits (line 911) stdout ----
error: multiple `cfg` predicates are specified
##[error] --> core/src/num/f16.rs:913:31
  |
5 | #[cfg(target_arch = "x86_64", target_os = "linux")] {

error: aborting due to 1 previous error

Couldn't compile the test.
Couldn't compile the test.
---- core/src/num/f16.rs - f16::f16::to_int_unchecked (line 868) stdout ----
error: multiple `cfg` predicates are specified
##[error] --> core/src/num/f16.rs:870:31
  |
5 | #[cfg(target_arch = "x86_64", target_os = "linux")] {

error: aborting due to 1 previous error

Couldn't compile the test.
```

</details>

--- a/library/core/src/num/f16.rs
+++ b/library/core/src/num/f16.rs
@@ -261,7 +261,7 @@ impl f16 {
     ///
     /// ```
     /// #![feature(f16)]
-    /// # #[cfg(target_arch = "aarch64")] { // FIXME(f16_F128): rust-lang/rust#123885
+    /// # #[cfg(all(target_arch = "x86_64", target_os = "linux"))] {
     ///
     /// let nan = f16::NAN;
     /// let f = 7.0_f16;
@@ -293,7 +293,7 @@ impl f16 {
     ///
     /// ```
     /// #![feature(f16)]
-    /// # #[cfg(target_arch = "aarch64")] { // FIXME(f16_F128): rust-lang/rust#123885
+    /// # #[cfg(all(target_arch = "x86_64", target_os = "linux"))] {
     ///
     /// let f = 7.0f16;
     /// let inf = f16::INFINITY;
@@ -319,7 +319,7 @@ impl f16 {
     ///
     /// ```
     /// #![feature(f16)]
-    /// # #[cfg(target_arch = "aarch64")] { // FIXME(f16_F128): rust-lang/rust#123885
+    /// # #[cfg(all(target_arch = "x86_64", target_os = "linux"))] {
     ///
     /// let f = 7.0f16;
     /// let inf: f16 = f16::INFINITY;
@@ -347,7 +347,7 @@ impl f16 {
     ///
     /// ```
     /// #![feature(f16)]
-    /// # #[cfg(target_arch = "aarch64")] { // FIXME(f16_F128): rust-lang/rust#123885
+    /// # #[cfg(all(target_arch = "x86_64", target_os = "linux"))] {
     ///
     /// let min = f16::MIN_POSITIVE; // 6.1035e-5
     /// let max = f16::MAX;
@@ -377,7 +377,7 @@ impl f16 {
     ///
     /// ```
     /// #![feature(f16)]
-    /// # #[cfg(target_arch = "aarch64")] { // FIXME(f16_F128): rust-lang/rust#123885
+    /// # #[cfg(all(target_arch = "x86_64", target_os = "linux"))] {
     ///
     /// let min = f16::MIN_POSITIVE; // 6.1035e-5
     /// let max = f16::MAX;
@@ -409,7 +409,7 @@ impl f16 {
     ///
     /// ```
     /// #![feature(f16)]
-    /// # #[cfg(target_arch = "aarch64")] { // FIXME(f16_F128): rust-lang/rust#123885
+    /// # #[cfg(all(target_arch = "x86_64", target_os = "linux"))] {
     ///
     /// use std::num::FpCategory;
     ///
@@ -867,7 +867,7 @@ impl f16 {
     ///
     /// ```
     /// #![feature(f16)]
-    /// # #[cfg(target_arch = "aarch64")] { // FIXME(f16_F128): rust-lang/rust#123885
+    /// # #[cfg(all(target_arch = "x86_64", target_os = "linux"))] {
     ///
     /// let value = 4.6_f16;
     /// let rounded = unsafe { value.to_int_unchecked::<u16>() };
@@ -910,7 +910,7 @@ impl f16 {
     ///
     /// ```
     /// #![feature(f16)]
-    /// # #[cfg(target_arch = "aarch64")] { // FIXME(f16_F128): rust-lang/rust#123885
+    /// # #[cfg(all(target_arch = "x86_64", target_os = "linux"))] {
     ///
     /// # // FIXME(f16_f128): enable this once const casting works
     /// # // assert_ne!((1f16).to_bits(), 1f16 as u128); // to_bits() is not casting!
@@ -958,7 +958,7 @@ impl f16 {
     ///
     /// ```
     /// #![feature(f16)]
-    /// # #[cfg(target_arch = "aarch64")] { // FIXME(f16_F128): rust-lang/rust#123885
+    /// # #[cfg(all(target_arch = "x86_64", target_os = "linux"))] {
     ///
     /// let v = f16::from_bits(0x4a40);
     /// assert_eq!(v, 12.5);
@@ -1071,7 +1071,7 @@ impl f16 {
     ///
     /// ```
     /// #![feature(f16)]
-    /// # #[cfg(target_arch = "aarch64")] { // FIXME(f16_F128): rust-lang/rust#123885
+    /// # #[cfg(all(target_arch = "x86_64", target_os = "linux"))] {
     ///
     /// let value = f16::from_be_bytes([0x4a, 0x40]);
     /// assert_eq!(value, 12.5);
@@ -1094,7 +1094,7 @@ impl f16 {
     ///
     /// ```
     /// #![feature(f16)]
-    /// # #[cfg(target_arch = "aarch64")] { // FIXME(f16_F128): rust-lang/rust#123885
+    /// # #[cfg(all(target_arch = "x86_64", target_os = "linux"))] {
     ///
     /// let value = f16::from_le_bytes([0x40, 0x4a]);
     /// assert_eq!(value, 12.5);
@@ -1124,7 +1124,7 @@ impl f16 {
     ///
     /// ```
     /// #![feature(f16)]
-    /// # #[cfg(target_arch = "aarch64")] { // FIXME(f16_F128): rust-lang/rust#123885
+    /// # #[cfg(all(target_arch = "x86_64", target_os = "linux"))] {
     ///
     /// let value = f16::from_ne_bytes(if cfg!(target_endian = "big") {
     ///     [0x4a, 0x40]
@@ -1257,7 +1257,7 @@ impl f16 {
     ///
     /// ```
     /// #![feature(f16)]
-    /// # #[cfg(target_arch = "aarch64")] { // FIXME(f16_F128): rust-lang/rust#123885
+    /// # #[cfg(all(target_arch = "x86_64", target_os = "linux"))] {
     ///
     /// assert!((-3.0f16).clamp(-2.0, 1.0) == -2.0);
     /// assert!((0.0f16).clamp(-2.0, 1.0) == 0.0);

